package mpsqual;

message ARINC429Request {
	enum RequestTypeT {
		DISCONNECT = 0;
		CONNECT	= 1;
		REPORT = 2;
	}
	
	required RequestTypeT requestType = 1 [default = DISCONNECT];
	required string sink = 2 [default = ""];
	optional string source = 3 [default = ""];
}

message ARINC429Response {
	enum ConStateT {
		DISCONNECTED = 0;
		CONNECTED = 1;
	}
	
	message StatusMessage {
		required string sink = 1;
		required string source = 2 [default = ""];
		required ConStateT conState = 3;
		required uint32 xmtCount = 4;
		required uint32 rcvCount = 5;
		required uint32 errorCount = 6;
	}

repeated StatusMessage status = 1;
}
